PocketKit/Sources/SharedPocketKit/Array+Extension.swift

```diff
@@ -13,3 +13,24 @@ extension Array {
         return self[index]
     }
 }
+
+extension Array where Element == CGFloat {
```

Comment:
I don't think this should be an extension of Array. Can we swap to be a helper function?

Reply:
curious to hear your perspective on why. The function is heavily tied to the array and acts as a method to the array. Not sure what the preference on the team to do standalone helper functions vs. extensions.

Reply:
I see both points here. On one hand, `getMaxHeightForRow` feels like a method that is specific for a `UITableView` (or `UICollectionView`). On the other, the operation is on an array.

My opinion is that we either add an helper or move this to where it's used and make it a `private` extension of `Array`.
Also, I'd ditch the `get` in the method name, but it's just a personal preference.

Reply:
If there is some core functionality to Array that we need in +50% of the app, then I think you have a good case for a public extension.
In this case, we only want this functionality in one or two spots, so does it make sense to make that available to the entire codebase?
In this case we are operating on an Array, yes. But it is an Array of heights, so you could say the functionality is more closely tied to Heights than to Array.

You could go so far as to make a happly little 
```
struct RowHeights {
    let heights: [CGFloat]

    func maxHeight() -> [CGFloat] {
        //Do the thing
    }
}
```

I do still have reservations around using Extensions as freely as we do. They obfuscate where code is and who wrote it.

Reply:
If I make it a private extension, then we won't be able rely on tests. Do we think a struct is over kill? I can convert it to a private function that takes in an array. But the operation being on an array seems more cleaner than injecting it. I have no strong opinions, so open to what's best for the team.

Reply:
you can make it a public extension, and maybe rename it in a more generic way?